R/estimate_uncertainty.R

```diff
@@ -41,40 +47,39 @@
 #'     82, 42, NA, NA,
 #'     70, NA, NA, NA
 #'   ),
-#'   nrow = 7,
+#'   nrow = 8,
 #'   byrow = TRUE
 #' )
 #'
-#' trunc_rts <- truncate_triangles(triangle, n = 2)
+#' trunc_rts <- truncate_triangles(triangle, n = 3)
 #' retro_rts <- construct_triangles(trunc_rts)
 #'
 #' retro_nowcasts <- fill_triangles(retro_rts, n = 5)
+#' # Estimate dispersion parameters using default (negative binomial error
+#' # model on the sums)
 #' disp_params <- estimate_uncertainty(
 #'   point_nowcast_matrices = retro_nowcasts,
 #'   truncated_reporting_triangles = trunc_rts,
-#'   retro_reporting_triangles = retro_rts,
-#'   n = 2
+#'   retro_reporting_triangles = retro_rts
 #' )
 #' disp_params
 #'
-#' # Estimate dispersion parameters from rolling sum
+#' # Estimate dispersion parameters from rolling mean with a normal error model
 #' disp_params_agg <- estimate_uncertainty(
 #'   point_nowcast_matrices = retro_nowcasts,
 #'   truncated_reporting_triangles = trunc_rts,
 #'   retro_reporting_triangles = retro_rts,
-#'   n = 2,
-#'   fun_to_aggregate = sum,
-#'   k = 2
+#'   ref_time_aggregator = function(x) zoo::rollsum(x, k = 2, align = "right")
 #' )
 #' disp_params_agg
 estimate_uncertainty <- function(
     point_nowcast_matrices,
     truncated_reporting_triangles,
     retro_reporting_triangles,
     n = length(point_nowcast_matrices),
-    fun_to_aggregate = sum,
-    k = 1) {
-  .validate_aggregation_function(fun_to_aggregate)
+    error_model = function(obs, pred) fit_by_horizon(obs = obs, pred = pred),
```

Comment:
fit_by_horizon should take an error model as an arg as it is in reality as fancy for loop. then we can begin to compose function inputs which is the point of this design.

Reply:
Yup have changed this (but am using `observation_model`)